src/main/java/io/rcktapp/api/handler/firehose/DeliveryStreamNameSpliterator.java

```diff
@@ -0,0 +1,55 @@
+package io.rcktapp.api.handler.firehose;
+
+import com.amazonaws.services.kinesisfirehose.AmazonKinesisFirehose;
+import com.amazonaws.services.kinesisfirehose.model.ListDeliveryStreamsRequest;
+import com.amazonaws.services.kinesisfirehose.model.ListDeliveryStreamsResult;
+
+import java.util.List;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class DeliveryStreamNameSpliterator implements Spliterator<String> {
+    private final AmazonKinesisFirehose firehoseClient;
+    private String last = null;
+
+    private DeliveryStreamNameSpliterator(AmazonKinesisFirehose firehoseClient) {
+        this.firehoseClient = firehoseClient;
+    }
+
+    static Stream<String> stream(AmazonKinesisFirehose firehoseClient) {
+        return StreamSupport.stream(new DeliveryStreamNameSpliterator(firehoseClient), false);
+    }
+
+
+    @Override
+    public boolean tryAdvance(Consumer<? super String> action) {
+        ListDeliveryStreamsResult page = firehoseClient.listDeliveryStreams(getRequest(last));
+        List<String> names = page.getDeliveryStreamNames();
+        last = names.stream().reduce((first, second) -> second).orElse(null);
+        names.forEach(action);
```

Comment:
This doesn't look like much but what goes in (forEach) is the consumer - that they passed - so we're feeding the names to the consumer this way.